CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 2.8.12)
 project(zlog)
 
+add_definitions(-DWITH_CACHE -DWITH_STATS)
```

Comment:
This is close. I think we want to have a cmake option so we can turn this on and off. Something like

```
option(WITH_CACHING "add caching" OFF)
if(WITH_CACHING)
  add_definitions(-DWITH_CACHE)
endif()
```